packages/@react-native-windows/codegen/Document.md

```diff
@@ -0,0 +1,386 @@
+calling `getConstants` in JavaScript gets the result of all these methods combined together.
+
+### REACT_EVENT
+
+`REACT_EVENT` implements an event.
+`REACT_EVENT` attaches to a `std::function` field.
+
+The syntax of `REACT_EVENT` is `REACT_EVENT(field, [optional]eventName, [optional]eventEmitterName)`.
+
+- `field`: the C++ field to store a JavaScript callback.
+- `eventName`: the event name visible to JavaScript. The default value is `L"field"`.
+- `eventEmitterName`: the JavaScript module event emitter name. If this argument is not set, the `eventEmitterName` in `REACT_MODULE` will be used.
+
+```c++
+REACT_EVENT(AnEvent)
+std::function<void(int, std::string const&)> AnEvent;
+```
+
+### REACT_FUNCTION
+
+`REACT_FUNCTION` defines a JavaScript implemented function.
+after a TurboModule is loaded,
+the attached field will be initialized with a value,
+calling this field executes some code registered in JavaScript.
+
+The syntax of `REACT_FUNCTION` is `REACT_FUNCTION(field, [optional]functionName, [optional]moduleName)`.
+
+- `field`: the C++ field to store a JavaScript function.
+- `functionName`: the JavaScript function name. The default value is `L"field"`.
+- `moduleName`: the JavaScript module event emitter name. If this argument is not set, the `moduleName` in `REACT_MODULE` will be used.
+
+```c++
+REACT_FUNCTION(AddValues, addValues, RandomModuleName)
+std::function<void(int a, int b)> AddValues;
+```
+
+```JavaScript
+import { NativeModules } from 'react-native';
+
+class RandomModuleName {
+    addValues(a, b) {
+        // do something
+    }
+}
+
+global.__fbBatchedBridge.registerLazyCallableModule('RandomModuleName', () => new RandomModuleName());
+```
+
+### Type Projections
+
+Here are a list of supported types in TurboModule.
+
+| TypeScript | Flow | C++ |
+| ---------- | ---- | --- |
+| `string` | `*` | `std::string` |
+| `number` | `*` | `double` |
+| `float (number)` | `*` | `double` |
+| `double (number)` | `*` | `double` |
+| `Int32 (number)` | `*` | `int` |
+| `boolean` | `*` | `bool` |
+| `T[]` | `*` | `std::vector<T>` |
+| `readonly T[]` | `$ReadOnlyArray<T>` | `std::vector<T>` |
+| `Array` | `*` | `winrt::Microsoft::ReactNative::JSValueArray` |
+| `object` | `*` | `winrt::Microsoft::ReactNative::JSValue` |
+| `{ literal-type }` | `{\| literal-type \|}` | A generated C++ struct |
+| `RootTag` | `*` | double |
+| `T` | `ReactNull` | `?T` | `std::optional<T>` |
+| `function (parameters) => void` | `(arguments) => void` | `std::function<void(parameters)>` |
+
+The following types are only valid for function return types
+
+| TypeScript | Flow | C++ |
+| ---------- | ---- | --- |
+| `void` | `*` | `void` |
+| `Promise<T>` | `*` | `winrt::Microsoft::ReactNative::ReactPromise<T>` (becomes the last argument) |
+
+**Important**:
+
+- `float`, `double`, `Int32` and `ReactNull` must be a type definition imported from another file.
```

Comment:
>another file [](http://example.com/codeflow?start=85&length=12)

Can we specify the file name from which they can be imported?

Reply:
Technically they can be imported from anywhere for people using `react-native-tscodegen`. Given that the `RN-TSCodegen-Types` is experimental, users could just make their own file.